helix-term/src/keymap.rs

```diff
@@ -62,252 +283,298 @@ impl DerefMut for Keymaps {
+                "j" => scroll_down
+            },
 
-            key!('"') => Command::select_register,
-        );
+            "\"" => select_register
+        });
         // TODO: decide whether we want normal mode to also be select mode (kakoune-like), or whether
         // we keep this separate select mode. More keys can fit into normal mode then, but it's weird
         // because some selection operations can now be done from normal mode, some from select mode.
         let mut select = normal.clone();
-        select.extend(
-            hashmap!(
-                key!('h') => Command::extend_char_left,
-                key!('j') => Command::extend_line_down,
-                key!('k') => Command::extend_line_up,
-                key!('l') => Command::extend_char_right,
-
-                key!(Left) => Command::extend_char_left,
-                key!(Down) => Command::extend_line_down,
-                key!(Up) => Command::extend_line_up,
-                key!(Right) => Command::extend_char_right,
-
-                key!('w') => Command::extend_next_word_start,
-                key!('b') => Command::extend_prev_word_start,
-                key!('e') => Command::extend_next_word_end,
-
-                key!('t') => Command::extend_till_char,
-                key!('f') => Command::extend_next_char,
-
-                key!('T') => Command::extend_till_prev_char,
-                key!('F') => Command::extend_prev_char,
-                key!(Home) => Command::goto_line_start,
-                key!(End) => Command::goto_line_end,
-                key!(Esc) => Command::exit_select_mode,
-            )
-            .into_iter(),
-        );
-
+        select.merge_nodes(keymap!({ "Select mode"
+            "h" | "left" => extend_char_left,
+            "j" | "down" => extend_line_down,
+            "k" | "up" => extend_line_up,
+            "l" | "right" => extend_char_right,
+
+            "w" => extend_next_word_start,
+            "b" => extend_prev_word_start,
+            "e" => extend_next_word_end,
+
+            "t" => extend_till_char,
+            "f" => extend_next_char,
+            "T" => extend_till_prev_char,
+            "F" => extend_prev_char,
+
+            "home" => goto_line_start,
+            "end" => goto_line_end,
+            "esc" => exit_select_mode
+        }));
+        let insert = keymap!({ "Insert mode"
+            "esc" => normal_mode,
+
+            "backspace" => delete_char_backward,
+            "del" => delete_char_forward,
+            "ret" => insert_newline,
+            "tab" => insert_tab,
+            "C-w" => delete_word_backward,
+
+            "left" => move_char_left,
+            "down" => move_line_down,
+            "up" => move_line_up,
+            "right" => move_char_right,
+            "pageup" => page_up,
+            "pagedown" => page_down,
+            "home" => goto_line_start,
+            "end" => goto_line_end_newline,
+
+            "C-x" => completion
```

Comment:
Can you please add additional `,` for the macro?

Reply:
Done 👍🏾